### PR TITLE
Revise staking

### PR DESCRIPTION
**Pull request by moreal**
# Changes

- **`ActionObsoleteConfig.V200080ObsoleteIndex` = 7_974_443L;**
- Introduce `StakePolicySheet`
- Limit `PatchTableSheet` action's behavior.
- Revise `Stake` action
- Revise `ClaimStakeReward` action

# Links

 - https://github.com/planetarium/NCIPs/pull/7

### CODE REVIEW

File: Lib9c/Action/Stake.cs [hunk 1]
```diff
@@ -104,13 +104,13 @@ public override IAccountStateDelta Execute(IActionContext context)
...
+                stakePolicySheet["StakeRegularFixedRewardSheet"].Value,
                 stakeRegularRewardSheetTableName:
-                    stakePolicySheet["StakeRegularRewardSheet"].Value,
+                stakePolicySheet["StakeRegularRewardSheet"].Value,
                 rewardInterval:
-                    long.Parse(stakePolicySheet["RewardInterval"].Value, CultureInfo.InvariantCulture),
+                long.Parse(stakePolicySheet["RewardInterval"].Value, CultureInfo.InvariantCulture),
                 lockupInterval:
-                    long.Parse(stakePolicySheet["LockupInterval"].Value, CultureInfo.InvariantCulture)
+                long.Parse(stakePolicySheet["LockupInterval"].Value, CultureInfo.InvariantCulture)
```
Comment (moreal) [inline Lib9c/Action/Stake.cs]: Is it intended? 👀

File: Lib9c/Action/Stake.cs [hunk 2]
```diff
@@ -132,39 +132,38 @@ public override IAccountStateDelta Execute(IActionContext context)
                 throw new StakeExistingClaimableException();
             }
 
-            // NOTE: Try cancel.
-            if (Amount == 0)
+            // NOTE: When the staking state is locked up.
```
Comment (moreal) [inline Lib9c/Action/Stake.cs:135]: 👍🏻 Good comments 👍🏻

File: Lib9c/Action/Stake.cs (hunk 1 above)
Comment (boscohyun) [reply Lib9c/Action/Stake.cs]: Oh, I didn't intended. It was automatically formatted. And.. it'll be changed.😉

File: Lib9c/TableData/Stake/StakePolicySheet.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        {
+            ("StakeRegularFixedRewardSheet", Contract.StakeRegularFixedRewardSheetPrefix),
+            ("StakeRegularRewardSheet", Contract.StakeRegularRewardSheetPrefix),
+        };
+
+        public string StakeRegularFixedRewardSheetValue =>
+            this["StakeRegularFixedRewardSheet"].Value;
+
+        public string StakeRegularRewardSheetValue =>
+            this["StakeRegularRewardSheet"].Value;
```
Comment (moreal) [inline Lib9c/TableData/Stake/StakePolicySheet.cs:65]: When knowing `ISheet.IRow.Value` property, the `Value` suffix seems good but it seems like having `StakeRegularFixedRewardSheet`'s value (i.e., CSV content). How do you think? @boscohyun

File: Lib9c/Model/State/GameConfigState.cs [hunk 4]
```diff
@@ -130,14 +172,50 @@ public override IValue Serialize()
...
+
 #pragma warning disable LAA1002
             return new Dictionary(values.Union((Dictionary) base.Serialize()));
 #pragma warning restore LAA1002
         }
 
         public void Set(GameConfigSheet sheet)
         {
-            foreach (var row in sheet.OrderedList)
+            foreach (var row in sheet.OrderedList!)
```
Comment (moreal) [inline Lib9c/Model/State/GameConfigState.cs:218]: This seems incorrect `nullable` type remover. How can it guarantee `Sheet.OrderedList` is not null at that time?

File: Lib9c/ActionObsoleteConfig.cs [hunk 5]
```diff
@@ -80,6 +80,9 @@ public static class ActionObsoleteConfig
 
         public const long V200071ObsoleteIndex = 7_718_878L;
 
+        // NOTE: This is temporary value.
+        public const long V200080ObsoleteIndex = 7_770_000L;
```
Comment (moreal) [inline Lib9c/ActionObsoleteConfig.cs:84]: We should delay this block index.

**Pull request merged**